### PR TITLE
[feature] support lazy init

### PR DESCRIPTION
**Pull request by ver217**
1. Add `pin_memory` arg for minimum waste search
2. Support lazy init

### CODE REVIEW

File: elixir/wrapper/module.py [hunk 1]
```diff
@@ -5,6 +5,7 @@
 
 import torch
 import torch.nn as nn
+from colossalai.utils.model.experimental import LazyTensor
```
Comment (1SAA) [inline elixir/wrapper/module.py:8]: @ver217 
Could you use the function `_lazy_init_check(param)` to check lazy init?

Comment (ver217) [reply elixir/wrapper/module.py:8]: done

Comment (1SAA) [reply elixir/wrapper/module.py:8]: ...There is one function in the module which is prepared for you.

Comment (ver217) [reply elixir/wrapper/module.py:8]: updated

**Comment by 1SAA** [comment]
Do you installed pre-commit? It seems like your format is not consistent with the format in this repo.

**Comment by ver217** [comment]
> Do you installed pre-commit? It seems like your format is not consistent with the format in this repo.

DONE

**Review by 1SAA — APPROVED** [review]

**Pull request merged**